src/app/leverage/provider.tsx

```diff
@@ -166,42 +130,32 @@ export function LeverageProvider(props: { children: any }) {
     return inputToken
   }, [inputToken, isMinting, outputToken])
 
+  const indexTokens = useMemo(() => {
+    return getLeverageTokens(chainId ?? 1)
+  }, [chainId])
+
+  const indexTokenAddresses = useMemo(() => {
+    return indexTokens.map((token) => token.address!)
+  }, [indexTokens])
+
+  const { balances, forceRefetchBalances } = useBalances(
+    address,
+    indexTokenAddresses,
+  )
+
   const indexTokenBasedOnLeverageType = useMemo(() => {
-    if (baseToken.symbol === 'ETH') {
-      switch (leverageType) {
-        case LeverageType.Long2x:
-          return IndexCoopEthereum2xIndex
-        case LeverageType.Long3x:
-          return IndexCoopEthereum3xIndex
-        case LeverageType.Short:
-          return IndexCoopInverseEthereumIndex
-      }
-    } else {
-      switch (leverageType) {
-        case LeverageType.Long2x:
-          return IndexCoopBitcoin2xIndex
-        case LeverageType.Long3x:
-          return IndexCoopBitcoin3xIndex
-        case LeverageType.Short:
-          return IndexCoopInverseBitcoinIndex
-      }
-    }
-  }, [baseToken, leverageType])
+    return indexTokens.filter(
+      (token) =>
+        token.baseToken === baseToken.symbol &&
+        token.leverageType === leverageType,
+    )[0]
```

Comment:
nit: could use `find` here instead https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

Reply:
#classic I was thinking about how it was called and then forgot to check later ;D